tests/functional/m/missing/missing_timeout.py

```diff
@@ -0,0 +1,85 @@
+"""Tests for missing-timeout."""
+
+# pylint: disable=consider-using-with,import-error,no-member,no-name-in-module,reimported
+
+import requests
+from requests import (
+    delete,
+    delete as delete_r,
+    get,
+    get as get_r,
+    head,
+    head as head_r,
+    options,
+    options as options_r,
+    patch,
+    patch as patch_r,
+    post,
+    post as post_r,
+    put,
+    put as put_r,
+    request,
+    request as request_r,
+)
+
+# requests without timeout
+requests.delete("http://localhost")  # [missing-timeout]
+requests.get("http://localhost")  # [missing-timeout]
+requests.head("http://localhost")  # [missing-timeout]
+requests.options("http://localhost")  # [missing-timeout]
+requests.patch("http://localhost")  # [missing-timeout]
+requests.post("http://localhost")  # [missing-timeout]
+requests.put("http://localhost")  # [missing-timeout]
+requests.request("call", "http://localhost")  # [missing-timeout]
+
+delete_r("http://localhost")  # [missing-timeout]
+get_r("http://localhost")  # [missing-timeout]
+head_r("http://localhost")  # [missing-timeout]
+options_r("http://localhost")  # [missing-timeout]
+patch_r("http://localhost")  # [missing-timeout]
+post_r("http://localhost")  # [missing-timeout]
+put_r("http://localhost")  # [missing-timeout]
+request_r("call", "http://localhost")  # [missing-timeout]
+
+delete("http://localhost")  # [missing-timeout]
+get("http://localhost")  # [missing-timeout]
+head("http://localhost")  # [missing-timeout]
+options("http://localhost")  # [missing-timeout]
+patch("http://localhost")  # [missing-timeout]
+post("http://localhost")  # [missing-timeout]
+put("http://localhost")  # [missing-timeout]
+request("call", "http://localhost")  # [missing-timeout]
+
+kwargs_wo_timeout = {}
+post("http://localhost", **kwargs_wo_timeout)
```

Comment:
This should ideally raise, but let's leave as is for now. This PR adds enough as is 😄

Reply:
I totally agree

To be honest I didn't know how to inspect a dictionary from astroid/pylint

If you can share me a example I can do it in a new PR

Reply:
I think some inspiration could be taken from the type checker (``missing-kwoa`` i.e. missing mandatory keyword argument in particular ) : https://github.com/PyCQA/pylint/blob/main/pylint/checkers/typecheck.py#L1379

Reply:
Thank you @Pierre-Sassoulas 

It worked like a charm

Also, It was committed